# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/PCookieStore.ipdl
# Commit: 0868fb4018a7
# Full Hash: 0868fb4018a7ae2002b8f2fcfd88643c2e060f07
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 9c8b6e35bf3f -r 0868fb4018a7 dom/cookiestore/PCookieStore.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/cookiestore/PCookieStore.ipdl	Tue Sep 10 16:31:38 2024 +0000
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include protocol PBackground;
+
+using mozilla::OriginAttributes from "mozilla/ipc/BackgroundUtils.h";
+using struct mozilla::void_t from "ipc/IPCMessageUtils.h";
+
+namespace mozilla {
+namespace dom {
+
+struct CookieData
+{
+  nsString name;
+  nsString value;
+};
+
+union MaybeCookieData
+{
+  CookieData;
+  void_t;
+};
+
+// This protocol is used for the CookieStore API
+[ManualDealloc]
+protocol PCookieStore
+{
+  manager PBackground;
+
+parent:
+  async GetRequest(nsString domain,
+                   OriginAttributes attrs,
+                   bool matchName,
+                   nsString name,
+                   nsCString path,
+                   bool onlyFirstMatch) returns (CookieData[] data);
+
+  async SetRequest(nsString domain,
+                   OriginAttributes attrs,
+                   nsString name,
+                   nsString value,
+                   bool session,
+                   int64_t expires,
+                   nsString path,
+                   int32_t sameSite,
+                   bool partitioned) returns (bool v);
+
+  async DeleteRequest(nsString domain,
+                      OriginAttributes attrs,
+                      nsString name,
+                      nsString path,
+                      bool partitioned) returns (bool v);
+
+  async Close();
+
+child:
+  async __delete__();
+};
+
+} // namespace dom
+} // namespace mozilla